lib/stripe/plans.rb

```diff
@@ -21,6 +21,36 @@ def initialize(*args)
       end
 
       def create_options
+        if api_version_after_plans_to_product_change
+          default_create_options
+        else
+          pre_product_create_options
+        end
+      end
+
+      def api_version_after_plans_to_product_change
+        plans_with_product_api_version = Date.parse('2018-02-05')
+
+        Stripe.api_version &&
+          Date.parse(Stripe.api_version) >= plans_with_product_api_version
+      end
```

Comment:
If you don't mind I have a suggestion on method naming here. It took me a few reads of the method name to understand what was happening here. Might I suggest the following refactor? You'll also note that it no longer needs the `plans_with_product_api_version` variable as the method name is clear enough to not need additional explanation via variable name. Also, as this method only returns a true or false, and is only used in a conditional, I end it with a question mark to make it clear that it returns a boolean value.

```
def api_version_gte_2018_02_05?
  Stripe.api_version && Date.parse(Stripe.api_version) >= Date.parse('2018-02-05')
end
```

Reply:
Good point about the conditional!

I am terrible with names! And I am glad you brought that up. However, I don't think `api_version_gte_2018_02_05` is a good fit either. Consider, if I was reading these lines of code,

```ruby
if api_version_gte_2018_02_05?
  default_create_options
else
  pre_product_create_options
end
```

I'll have to wonder: what happened after 2018-02-05?

```ruby
if api_version_after_plans_to_product_change?
  default_create_options
else
  pre_product_create_options
end
```

This is also terrible! But gives at least some idea of what happened.

```ruby
if api_version_after_switch_to_products_in_plans?
  default_create_options
else
  create_options_without_products
end
```

This might be better.